blockindex/contractstaking/dirty_cache.go

```diff
@@ -105,7 +105,7 @@ func (dirty *contractStakingDirty) finalize() (batch.KVStoreBatch, *contractStak
 
 func (dirty *contractStakingDirty) finalizeBatch() batch.KVStoreBatch {
 	dirty.once.Do(func() {
-		total := dirty.clean.TotalBucketCount() + dirty.delta.AddedBucketCnt()
+		total := dirty.clean.totalBucketCount + dirty.delta.AddedBucketCnt()
```

Comment:
not needed?
TotalBucketCount() has a lock inside

Reply:
ok, use 0 height to query latest state